### PR TITLE
Fix #78: unresolved dependency '__path__' in '__init__.py'

### LINKED ISSUE #78: Bug: unresolved dependency '__path__' in __init__.py
Consider the following package
```bash
> tree pkg
pkg
└── __init__.py

0 directories, 1 file
```
where `__init__.py` contains the following code:
```bash
print(__path__)
```

This package is a valid python package and the following code runs:
```python
> python -c "import pkg"
['/home/<user>/ssort/pkg']
```

However, if I run `ssort` in the `pkg` folder, I get the following error:
```bash
> ssort
ERROR: unresolved dependency '__path__' in __init__.py: line 1, column 6
1 file was not sortable
```

This should not happen since `__path__` is defined in the `__init__.py` file of a package.

Instead, consider the package
```bash
pkg
├── file.py
└── __init__.py

0 directories, 2 files
```
with
```python
# __init__.py
from . import file

# file.py
print(__path__)
```
Then the following code fails
```bash
> python -c "import pkg"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/<user>/ssort/pkg/__init__.py", line 2, in <module>
    from . import file
  File "/home/<user>/ssort/pkg/file.py", line 2, in <module>
    print(__path__)
          ^^^^^^^^
NameError: name '__path__' is not defined. Did you mean: '__name__'?
```
and `ssort` rightfully complains
```bash
> ssort
ERROR: unresolved dependency '__path__' in pkg/file.py: line 2, column 6
1 file was left unchanged, 1 file was not sortable
```

Altogether, the first case is a bug.

TL, DR: in the `__init__.py` file of a package the variable `__path__` is defined.

### PR DESCRIPTION
**Pull request by cgahr**
Fix #78

### CODE REVIEW

File: src/ssort/_dependencies.py [hunk 1]
```diff
@@ -44,6 +49,13 @@ def module_statements_graph(statements, *, on_unresolved, on_wildcard_import):
                 resolved[requirement] = None
                 continue
 
+            if (
+                filename.endswith("__init__.py")
```
Comment (bwhmather) [inline src/ssort/_dependencies.py:53]: An unlikely edge case, but we probably want to handle filenames such as `something__init__.py`.

**Comment by bwhmather** [comment]
Is there something we could have added to `test_builtins.py` that would have caught this?

**Comment by bwhmather** [comment]
I think @jgberry wins this one with https://github.com/bwhmather/ssort/pull/85.  I think he's right that ssort already has many false negatives, and that this isn't really a problem.

**Comment by jgberry** [comment]
> Is there something we could have added to `test_builtins.py` that would have caught this?

#85 also addresses this by adding a number of new test cases to `test_builtins.py`. These tests revealed we are also missing the following builtins in addition to `__path__`:
- `__annotations__`
- `__builtins__`
- `__cached__`

**Comment by cgahr** [comment]
Superseded by #85

**Pull request closed without merging**